Add `debug_traceTransaction` RPC method (#1328)
nomiclabs/hardhat: docs/hardhat-network/README.md

This logging is enabled by default when using Hardhat Network's node (i.e. `npx hardhat node`), but disabled when using
the in-process Hardhat Network provider. See [Hardhat Network's config](../config/README.md#hardhat-network) to learn more about how to control its logging.

## The `debug_traceTransaction` method

You can get debug traces of already mined transactions using the
`debug_traceTransaction` RPC method. The returned object has a detailed
description of the transaction execution, including a list of steps describing
each executed opcode and the state of the EVM at that point.

To get a trace, call this method with the hash of the transaction as its
argument:

```js
const trace = await hre.network.provider.send("debug_traceTransaction",
["0x123..."])
```

You can also selectively disable some properties in the list of steps:

```js
const trace = await hre.network.provider.send("debug_traceTransaction", [
  "0x123...",
  {
    disableMemory: true,
    disableStack: true,
    disableStorage: true,
  }
]);
```

> If you are using [mainnet forking](https://hardhat.org/guides/mainnet-forking.html) with an archive node,

Reviewer: ```suggestion
If you are using [the forking functionality](https://hardhat.org/guides/mainnet-forking.html) with an archive node,
```

> you can get traces of transactions from the mainnet even if the node you are using

Reviewer: ```suggestion
you can get traces of transactions from the remote network even if the node you are using
```

doesn't support
`debug_traceTransaction`.

> There are some known limitations around this method:

Reviewer: ```suggestion
### Known limitations
```
